plugins/modules/proxmox_disk.py

```diff
@@ -554,6 +554,8 @@ def create_disk(self, disk, vmid, vm, vm_config):
                 ok_str = "CD-ROM was created on %s bus in VM %s"
             else:
                 config_str = self.module.params["storage"]
+                if not config_str:
+                    self.module.fail_json(msg="The storage option must be specified.")
```

Comment:
Given the `required_if` with `(create forced [storage])`, I assume that `storage` is also going to be needed for `create=regular` and there is no `import_string` nor `iso_image`. I cannot vouch for that logical condition, but from a code point of view, this one LGTM.

That being said, this function seems bloated and in need of some refactoring into smaller pieces, IMHO.

Reply:
> That being said, this function seems bloated and in need of some refactoring into smaller pieces, IMHO.

I fully agree.